### ISSUE TITLE
Frequent DB connection timeout

### ISSUE THREAD

**Issue by yomexzo** [balderdashy/sails#2510]
I have been having issues with my sails app deployment. I often get the following stack trace saying sails-mysql / waterline experiences a timeout while trying to spawn a new connection. But in every case, the DB is always up and running as I can connect to it using mysql clients. It happens often but haven't been able to determine it's pattern or what triggers it.

	Error: connect ETIMEDOUT
	    at afterwards (/sites/smser/node_modules/sails-mysql/lib/connections/spawn.js:72:13)
	    at /sites/smser/node_modules/sails-mysql/lib/connections/spawn.js:40:7
	    at Pool.<anonymous> (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/Pool.js:47:16)
	    at Handshake.Sequence.end (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/protocol/sequences/Sequence.js:78:24)
	    at Protocol.handleNetworkError (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/protocol/Protocol.js:282:14)
	    at PoolConnection.Connection._handleNetworkError (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/Connection.js:303:18)
	    at PoolConnection.Connection._handleConnectTimeout (/sites/smser/node_modules/sails-mysql/node_modules/mysql/lib/Connection.js:299:8)
	    at Socket.g (events.js:180:16)
	    at Socket.emit (events.js:92:17)
	    at Socket._onTimeout (net.js:327:8)
	Details:  Error: Could not connect to MySQL:
	Error: connect ETIMEDOUT

**Comment by yomexzo** [comment]
Yes it is. But i frequently get the "Error: ER_CON_COUNT_ERROR: Too many connections" error. I still don't understand why using pools don't work too.

**Comment by RWOverdijk** [comment]
:+1: currently facing the same oddities.

**Comment by CWyrtzen** [comment]
Do the oddities remain?

**Comment by malikba** [comment]
Hi,

I'm currently facing the same `ETIMEDOUT` error. My Sails instance is on an Azure VM and MySQL is on another VM. Anyone solved this issue ?

**Comment by sailsbot** [comment]
Thanks for posting, @yomexzo.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by wisquimas** [comment]
Sails´s Support sucks

**Comment by wulfsolter** [comment]
Sails's support rocks!!!! - you get what you pay for! http://sailsjs.com/flagship

I'm not affiliated with Sails crew in any way, just see the pattern of n00bs asking questions, n00bs don't bother answering when they solve it, and then more people post **different questions** expecting even more answers immediately and for free. It is really demoralizing to developers to have threads like these - getting bagged out because newbies haven't learnt about databases and connections.

I believe the support pricing is very competitive, and if you were to engage, I have no reason to believe every different issue in this thread would get solved.

But as it stands now, all I see is multiple different problems lumped under one thread, and people wanting something for free? Like they owe you something because you downloaded and installed someone else's hard work? Sheesh...
